src/momento/aio/_scs_data_client.py

```diff
@@ -78,10 +171,41 @@ async def get(self, cache_name: str, key: str) -> cache_sdk_resp.CacheGetRespons
                 timeout=self._default_deadline_seconds,
             )
             _momento_logger.debug(f"Received a get response for {key}")
-            return cache_sdk_resp.CacheGetResponse(response)
+            return cache_sdk_ops.CacheGetResponse(response)
         except Exception as e:
             _momento_logger.debug(f"Get failed for {key} with response: {e}")
             raise _cache_service_errors_converter.convert(e)
 
+    async def multi_get(
+        self,
+        cache_name: str,
+        get_operations: List[cache_sdk_ops.CacheMultiGetOperation],
+    ) -> cache_sdk_ops.CacheMultiGetResponse:
+        _validate_multi_op_list(get_operations)
+        _validate_cache_name(cache_name)
+        try:
+            rsp_promises = set()
+            for op in get_operations:
+                get_request = _GetRequest()
+                get_request.cache_key = _as_bytes(op.key, "Unsupported type for key: ")
+                rsp_promises.add(
+                    self._grpc_manager.async_stub().Get(
+                        get_request, metadata=_make_metadata(cache_name)
+                    )
+                )
+
+            responses = []
+            for pending_response in rsp_promises:
+                # await one at a time to keep ordered for user
+                completed_request = await pending_response
+                # Receive the returned result of current user
+                responses.append(cache_sdk_ops.CacheGetResponse(completed_request))
```

Comment:
@kvcache I was doing similar to this before but I wasn't totally sure from docs if `gather` was guaranteed to return promises in order they were submited especially in failure scenario even if I pass return_exceptions=true like I am now. So wasnt sure if associating the get_operations w/ request promises via zip like this would get a mismatch? That was why had switched to call back approach instead so that can explicitly pass in the associated request info into the call back and take action. 

Im reading docs some more this morning and still not totally clear to me if task fails will errors be ordered in nicely with results if `return_exceptions=true`. I can switch back to this for now and try and add some more unit testing around this try make sure is ok. Seems like you are interpreting docs as saying everything will be ordered nicely no matter what if we set the `return_exceptions` and so this approach of associating requests w/ futures is safe and cleaner. I definitely prefer this for readability as well code wise.